mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1001,3 +1115,100 @@ func decodeJSONFile(mt *mtest.T, file string, val interface{}) bson.Raw {
 func rawValueToCoreValue(rv bson.RawValue) bsoncore.Value {
 	return bsoncore.Value{Type: rv.Type, Data: rv.Value}
 }
+
+type deadlockTest struct {
+	clientTest           *mongo.Client
+	clientKeyVaultOpts   *options.ClientOptions
+	clientKeyVaultEvents []startedEvent
+	clientEncryption     *mongo.ClientEncryption
+	ciphertext           primitive.Binary
+}
+type startedEvent struct {
+	cmd string
+	db  string
+}
+
+func compareStarted(mt *mtest.T, clientName string, actual []startedEvent, expected []startedEvent) {
+	mt.Helper()
+	matched := false
+	defer func() {
+		if !matched {
+			mt.Fatalf("mismatched events for %q. Expected %v, got %v", clientName, expected, actual)
+		}
+	}()
+
+	if len(actual) != len(expected) {
+		return
+	}
+
+	for i, e := range expected {
+		if actual[i] != e {
+			return
+		}
+	}
+
+	matched = true
+}
+
+func newDeadlockTest(mt *mtest.T) *deadlockTest {
+	mt.Helper()
+
+	d := deadlockTest{}
```

Comment:
nit: Empty declarations can be done as `var d deadlockTest`.

Reply:
Done.